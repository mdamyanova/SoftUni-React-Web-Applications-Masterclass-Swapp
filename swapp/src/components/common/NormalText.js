import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'rebass';

const NormalText = ({ text, className, style }) => {
  return (
    <Text
      p={2}
      className={className}
      sx={{ ...style }}
      css={`
        color: var(--default-font-color);
      `}
    >
      {text}
    </Text>
  );
};

NormalText.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  style: PropTypes.object
};

NormalText.defaultProps = {
  className: '',
  style: {}
};

export default NormalText;
