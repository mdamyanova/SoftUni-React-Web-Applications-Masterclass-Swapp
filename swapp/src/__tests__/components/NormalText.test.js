import React from "react";
import { shallow, mount } from "enzyme";
import NormalText from "../../components/common/forms/NormalText";

describe("<NormalText />", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<NormalText className="testText" text="Test Normal Text" />)
    ).toMatchSnapshot();
  });

  it("should have props", () => {
    const wrapper = mount(
        <NormalText className="testText" text="Test Normal Text" />
    );
    expect(wrapper.props().className).toEqual("testText");
    expect(wrapper.props().text).toEqual("Test Normal Text");
  });
});
