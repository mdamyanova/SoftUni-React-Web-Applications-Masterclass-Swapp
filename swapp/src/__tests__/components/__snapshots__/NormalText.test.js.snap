// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NormalText /> should match snapshot 1`] = `
<ForwardRef
  className="testText"
  css="
        color: var(--default-font-color);
      "
  p={2}
  sx={Object {}}
>
  Test Normal Text
</ForwardRef>
`;
